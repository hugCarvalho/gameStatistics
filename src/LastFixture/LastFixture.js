import React from "react";
import "./LastFixture.scss";
import matchReport from "../data/matchReports";
// import teamStats from "./stats/statsTeams";

function LastFixture() {
  const lastMatch = matchReport[matchReport.length - 1];

  return (
    <header className="App-header">
      <h1>Baseball Highlights Statistics</h1>
      <h3>
        Last fixture: <time dateTime="2020-08-30">{lastMatch.date}</time>
      </h3>
      <section>
        <div className="wrapper__all">
          <div className="info">
            <span>{lastMatch.playerA.name}</span>
            <span>{lastMatch.playerA.team}</span>
            <span>{lastMatch.playerA.score}</span>
          </div>
          <div> vs </div>
          <div className="info">
            <span>{lastMatch.playerB.name}</span>
            <span>{lastMatch.playerB.team}</span>
            <span>{lastMatch.playerB.score}</span>
          </div>
        </div>
      </section>
    </header>
  );
}

export default LastFixture;
