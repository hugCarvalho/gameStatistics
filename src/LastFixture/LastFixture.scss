.App-header {
  color: rgb(228, 225, 225);
  section {
    border: 1px solid whitesmoke;
    width: 50%;
    margin: 30px auto 0 auto;

    border-radius: 40px;
    min-width: 700px;
    margin-bottom: 100px;
  }
  .wrapper__all {
    display: flex;
    align-items: center;
    justify-content: center;
    font-size: 2.4rem;
  }
  .info {
    // background-color: limegreen;
    flex-shrink: 0;
    display: flex;
    flex-direction: column;
    padding: 60px 30px;
    line-height: 4rem;
    min-width: 260px;
  }

  .info span:first-of-type {
    font-size: 1.4rem;
  }
}
