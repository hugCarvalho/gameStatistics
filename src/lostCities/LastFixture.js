import React from "react";
import formatDate from "./formatDate";
import "./LastFixture.scss";

function LastFixture({ matchesDatabase }) {
  return (
    <section className="LastFixture">
      <h2>Last Fixture</h2>
      <time>Date: {matchesDatabase.games[0].date && formatDate(matchesDatabase)}</time>
      <table className="greenTable">
        <thead>
          <tr>
            <th></th>
            <th>
              {matchesDatabase.games[0]
                ? matchesDatabase.games.reverse()[0].playerA.name // !!! Reverses in place !!!
                : ""}
            </th>
            <th>
              {matchesDatabase.games[0] ? matchesDatabase.games[0].playerB.name : ""}
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>Round 1</td>
            <td>
              {matchesDatabase.games[0] &&
                `${matchesDatabase.games[0].playerA.rounds[0]}`}
            </td>
            <td>
              {matchesDatabase.games[0] &&
                `${matchesDatabase.games[0].playerB.rounds[0]}`}
            </td>
          </tr>
          <tr>
            <td>Round 2</td>
            <td>
              {" "}
              {matchesDatabase.games[0] &&
                `${matchesDatabase.games[0].playerA.rounds[1]}`}
            </td>
            <td>
              {matchesDatabase.games[0] &&
                `${matchesDatabase.games[0].playerB.rounds[1]}`}
            </td>
          </tr>
          <tr>
            <td>Round_3</td>
            <td>
              {" "}
              {matchesDatabase.games[0] &&
                `${matchesDatabase.games[0].playerA.rounds[2]}`}
            </td>
            <td>
              {matchesDatabase.games[0] &&
                `${matchesDatabase.games[0].playerB.rounds[2]}`}
            </td>
          </tr>
          <tr>
            <td>Total</td>
            <td>{matchesDatabase.games[0] && matchesDatabase.games[0].playerA.total}</td>
            <td>{matchesDatabase.games[0] && matchesDatabase.games[0].playerB.total}</td>
          </tr>
        </tbody>
      </table>
    </section>
  );
}

export default LastFixture;
