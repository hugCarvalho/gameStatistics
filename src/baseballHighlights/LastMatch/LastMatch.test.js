import React from "react";
import renderer from "react-test-renderer";

import LastFixture from "./LastFixture";

it("renders correctly", () => {
  const tree = renderer.create(<LastFixture />).toJSON();
  expect(tree).toMatchSnapshot();
});
