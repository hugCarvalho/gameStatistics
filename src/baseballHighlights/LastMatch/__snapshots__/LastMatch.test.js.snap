// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<header
  className="App-header"
>
  <h1>
    Baseball Highlights Statistics
  </h1>
  <h3>
    "No matches have been played yet"
  </h3>
</header>
`;
